Yank/repex.py

```diff
@@ -2001,13 +1978,13 @@ def create(self, thermodynamic_states, sampler_states, storage, unsampled_thermo
         # Reset statistics.
         # _n_accepted_matrix[i][j] is the number of swaps proposed between thermodynamic states i and j.
         # _n_proposed_matrix[i][j] is the number of swaps proposed between thermodynamic states i and j.
-        self._n_accepted_matrix = np.zeros([self.n_replicas, self.n_replicas], np.int64)
-        self._n_proposed_matrix = np.zeros([self.n_replicas, self.n_replicas], np.int64)
+        self._n_accepted_matrix = np.zeros([self.n_states, self.n_states], np.int64)
```

Comment:
> This should still work in the general case for both Repex and SAMS since its Thermodynamic State <-> Thermodynamic State swaps yes?

Yes.

> A future check for SAMS (especially if local swaps are used) is to make sure that if the denominator when computing this is 0 to not throw an error (i.e. no swaps attempted)

I don't think we throw an error in this case now, so I'm not sure I understand what behavior change you are suggesting.

Reply:
> I don't think we throw an error in this case now,

Good point, I forgot we already have a "in case of divide by zero" check